File: coding-challenges/sources/daily-coding-problem/Day 12/Day12.kt
fun main() {
    println(climbStaircase(4) == 5)
}

fun climbStaircase(steps: Int): Int {
    if (steps < 2) return 1
    
    var ways = IntArray(steps + 1) { 1 }
    for (i in 2 until ways.size) {
        ways[i] = ways[i - 1] + ways[i - 2]
    }
    return ways[steps]
}

File: coding-challenges/sources/daily-coding-problem/Day 12/Day12.swift


fileprivate func climbStaircase(steps: Int) -> Int {
    if steps < 2 { return 1 }

    var ways = [Int](repeating: 1, count: steps + 1)
    for i in 2..<ways.count {
        ways[i] = ways[i - 1] + ways[i - 2]
    }
    return ways[steps]
}

import XCTest

final class Day12: XCTestCase {
    func test() {
        XCTAssertEqual(climbStaircase(steps: 4), 5)
    }
}
